bundles/com.espressif.idf.core/src/com/espressif/idf/core/toolchain/ESPToolChainManager.java

```diff
@@ -475,4 +458,39 @@ protected String getIdfCMakePath(String idfPath)
 	{
 		return idfPath + IPath.SEPARATOR + IDFConstants.TOOLS_FOLDER + IPath.SEPARATOR + IDFConstants.CMAKE_FOLDER;
 	}
+
+	/**
+	 * Configure the file in the preferences and initialize the launch bar with available targets
+	 */
+	public void configureToolChain()
+	{
+		IToolChainManager tcManager = CCorePlugin.getService(IToolChainManager.class);
+		ICMakeToolChainManager cmakeTcManager = CCorePlugin.getService(ICMakeToolChainManager.class);
+
+		initToolChain(tcManager, ESPToolChainProvider.ID);
+		initCMakeToolChain(tcManager, cmakeTcManager);
+		addToolchainBasedTargets(IDFCorePlugin.getService(ILaunchTargetManager.class));
+	}
+
+}
+
+class ESPToolChainElement
+{
+	public String name;
+	public String id;
+	public String arch;
+	public String fileName;
+	public String compilerPattern;
+	public String debuggerPattern;
```

Comment:
since these values ​​are public, we must make them static final constants. Also, maybe we should introduce one more field here 'typeId' which will separate the gcc from the clang toolchains. For now, we have only gcc toolchains defined in the extension point

Reply:
We can't make static final considering values are assigned in the constructor

Reply:
> We can't make static final considering values are assigned in the constructor.

Hi @kolipakakondal, sorry for the confusion. I meant the final constant so we can't change the value after the object is created